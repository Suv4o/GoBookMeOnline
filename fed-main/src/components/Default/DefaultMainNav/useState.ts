import { reactive, toRefs } from 'vue'

const state = reactive({
  isSignInButtonShown: true,
  isSignUpButtonShown: true,
})

export default () => {
  function setSignInButtonShown(isShown = true) {
    state.isSignInButtonShown = isShown
  }

  function setSignUpButtonShown(isShown = true) {
    state.isSignUpButtonShown = isShown
  }

  return {
    ...toRefs(state),
    setSignInButtonShown,
    setSignUpButtonShown,
  }
}
