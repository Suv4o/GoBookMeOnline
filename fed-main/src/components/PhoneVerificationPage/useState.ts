import { reactive, toRefs } from 'vue'

const state = reactive({
  fullName: '',
  phoneNumber: '',
})

export default () => {
  function setFullName(name = '') {
    state.fullName = name
  }

  function setPhoneNumber(phone = '') {
    state.phoneNumber = phone
  }

  return {
    ...toRefs(state),
    setFullName,
    setPhoneNumber,
  }
}
