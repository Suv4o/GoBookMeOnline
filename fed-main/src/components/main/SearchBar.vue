<script lang="ts">
export default {
  name: 'SearchBar',
}
</script>

<script setup lang="ts">
import SearchInput from './SearchInput.vue'
import LocationInput from './LocationInput.vue'
</script>

<template>
  <div class="flex flex-row mx-auto max-w-3xl">
    <div class="basis-2/3 mr-4">
      <search-input />
    </div>
    <div class="basis-1/3"><location-input /></div>
  </div>
</template>
