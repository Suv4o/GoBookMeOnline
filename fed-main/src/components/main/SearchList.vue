<script lang="ts">
export default {
  name: 'SearchList',
}
</script>

<script setup lang="ts">
import BookSearchList from '../global/BookSearchList/BookSearchList.vue'
import BookPagination from '../global/BookPagination/BookPagination.vue'
</script>

<template>
  <book-search-list title="Locations" type="location" />
  <book-search-list title="People" type="person" />
  <div class="mt-4">
    <book-pagination />
  </div>
</template>
