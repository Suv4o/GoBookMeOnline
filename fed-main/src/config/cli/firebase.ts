#!/usr/bin/env node
import * as yargs from 'yargs'
import * as dotenv from 'dotenv'
import * as admin from 'firebase-admin'
const env = dotenv.config({ path: `./env/.env.${process.env.NODE_ENV}` })

const userIds = [] as string[]

const app: admin.app.App = admin.initializeApp({
  credential: admin.credential.cert({
    type: env.parsed?.VITE_FIREBASE_TYPE,
    project_id: env.parsed?.VITE_FIREBASE_PROJECT_ID,
    private_key_id: env.parsed?.VITE_FIREBASE_PRIVATE_KEY_ID,
    private_key: Buffer.from(env.parsed?.VITE_FIREBASE_PRIVATE_KEY || '', 'base64').toString(),
    client_email: env.parsed?.VITE_FIREBASE_CLIENT_EMAIL,
    client_id: env.parsed?.VITE_FIREBASE_CLIENT_ID,
    auth_uri: env.parsed?.VITE_FIREBASE_AUTH_URI,
    token_uri: env.parsed?.VITE_FIREBASE_TOKEN_URI,
    auth_provider_x509_cert_url: env.parsed?.VITE_FIREBASE_AUTH_PROVIDER_X509_CERT_URL,
    client_x509_cert_url: env.parsed?.VITE_FIREBASE_CLIENT_X509_CERT_URL,
  } as any),
})

const getAllUsersIds = async (nextPageToken?: string) => {
  try {
    const listUsersResult = await app.auth().listUsers(1000, nextPageToken)
    listUsersResult.users.forEach(userRecord => {
      userIds.push(userRecord.uid)
    })
    if (listUsersResult.pageToken) {
      getAllUsersIds(listUsersResult.pageToken)
    }
  } catch (error) {
    console.log('Error listing users:', error)
  }
}

yargs.command(
  'drop',
  'firebase drop users',
  async () => {
    await getAllUsersIds()
    if (userIds.length) {
      try {
        const deleteUsersResult = await app.auth().deleteUsers(userIds)
        console.log(`Successfully deleted ${deleteUsersResult.successCount} users`)
        console.log(`Failed to delete ${deleteUsersResult.failureCount} users`)
      } catch (error) {
        console.log('Error deleting users:', error)
      }
    } else {
      console.log('No users to delete')
    }
  },
  argv => {
    console.log(argv)
  }
).argv
