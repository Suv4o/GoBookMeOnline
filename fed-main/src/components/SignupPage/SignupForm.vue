<script lang="ts">
export default {
  name: 'SignupForm',
}
</script>

<script setup lang="ts">
import { signInWithPopup, GoogleAuthProvider } from 'firebase/auth'
import { Auth } from '@firebase/auth'
import { Assertions } from '../../types/guards'
import { inject } from 'vue'
import { FirebaseUserResponse } from '../../types/interfaces'
import { useFetch } from '../../utils/composables/fetch'

const googleProvider = new GoogleAuthProvider()
const $auth = inject('$auth') as Auth

async function signInWithGoogle() {
  try {
    await signInWithPopup($auth, googleProvider)
    storeUserToDatabase()
  } catch (error) {
    Assertions.isFirebaseError(error)
    const code = error.code
    const message = error.message
    const credential = GoogleAuthProvider.credentialFromError(error)
    console.error(`Code: ${code}, Message: ${message}, Credential: ${credential}`)
  }
}

async function storeUserToDatabase() {
  const { error, data, isLoading } = await useFetch({
    url: '/user/signup-with-provider',
    method: 'POST',
    credentials: true,
  })
}
</script>

<template>
  <div class="bg-white sm:max-w-md sm:w-full sm:mx-auto sm:rounded-lg sm:overflow-hidden">
    <div class="px-4 py-8 sm:px-10">
      <div>
        <p class="text-sm font-medium text-gray-700">Sign in with</p>

        <div class="mt-1 grid grid-cols-3 gap-3">
          <div>
            <a
              href="#"
              class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
            >
              <span class="sr-only">Sign in with Facebook</span>
              <svg class="w-5 h-5" aria-hidden="true" fill="currentColor" viewBox="0 0 20 20">
                <path
                  fill-rule="evenodd"
                  d="M20 10c0-5.523-4.477-10-10-10S0 4.477 0 10c0 4.991 3.657 9.128 8.438 9.878v-6.987h-2.54V10h2.54V7.797c0-2.506 1.492-3.89 3.777-3.89 1.094 0 2.238.195 2.238.195v2.46h-1.26c-1.243 0-1.63.771-1.63 1.562V10h2.773l-.443 2.89h-2.33v6.988C16.343 19.128 20 14.991 20 10z"
                  clip-rule="evenodd"
                />
              </svg>
            </a>
          </div>

          <div>
            <a
              href="javascript:;"
              class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
              @click="signInWithGoogle"
            >
              <span class="sr-only">Sign in with Google</span>
              <svg class="w-5 h-5" aria-hidden="true" fill="currentColor" viewBox="0 0 210 210">
                <path
                  d="M0,105C0,47.103,47.103,0,105,0c23.383,0,45.515,7.523,64.004,21.756l-24.4,31.696C133.172,44.652,119.477,40,105,40
	c-35.841,0-65,29.159-65,65s29.159,65,65,65c28.867,0,53.398-18.913,61.852-45H105V85h105v20c0,57.897-47.103,105-105,105
	S0,162.897,0,105z"
                />
              </svg>
            </a>
          </div>

          <div>
            <a
              href="#"
              class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
            >
              <span class="sr-only">Sign in with Apple</span>
              <svg class="w-5 h-5" aria-hidden="true" fill="currentColor" viewBox="0 0 512.003 512.003">
                <g>
                  <g>
                    <path
                      d="M351.98,0c-27.296,1.888-59.2,19.36-77.792,42.112c-16.96,20.64-30.912,51.296-25.472,81.088
			c29.824,0.928,60.64-16.96,78.496-40.096C343.916,61.568,356.556,31.104,351.98,0z"
                    />
                  </g>
                </g>
                <g>
                  <g>
                    <path
                      d="M459.852,171.776c-26.208-32.864-63.04-51.936-97.824-51.936c-45.92,0-65.344,21.984-97.248,21.984
			c-32.896,0-57.888-21.92-97.6-21.92c-39.008,0-80.544,23.84-106.88,64.608c-37.024,57.408-30.688,165.344,29.312,257.28
			c21.472,32.896,50.144,69.888,87.648,70.208c33.376,0.32,42.784-21.408,88-21.632c45.216-0.256,53.792,21.92,87.104,21.568
			c37.536-0.288,67.776-41.28,89.248-74.176c15.392-23.584,21.12-35.456,33.056-62.08
			C387.852,342.624,373.932,219.168,459.852,171.776z"
                    />
                  </g>
                </g>
              </svg>
            </a>
          </div>
        </div>
      </div>

      <div class="mt-6 relative">
        <div class="absolute inset-0 flex items-center" aria-hidden="true">
          <div class="w-full border-t border-gray-300" />
        </div>
        <div class="relative flex justify-center text-sm">
          <span class="px-2 bg-white text-gray-500"> Or </span>
        </div>
      </div>

      <div class="mt-6">
        <form action="#" method="POST" class="space-y-6">
          <div>
            <label for="name" class="sr-only">Full name</label>
            <input
              id="name"
              type="text"
              name="name"
              autocomplete="name"
              placeholder="Full name"
              required="true"
              class="block w-full shadow-sm focus:ring-teal-500 focus:border-teal-500 sm:text-sm border-gray-300 rounded-md"
            />
          </div>

          <div>
            <label for="mobile-or-email" class="sr-only">Mobile number or email</label>
            <input
              id="mobile-or-email"
              type="text"
              name="mobile-or-email"
              autocomplete="email"
              placeholder="Mobile number or email"
              required="true"
              class="block w-full shadow-sm focus:ring-teal-500 focus:border-teal-500 sm:text-sm border-gray-300 rounded-md"
            />
          </div>

          <div>
            <label for="password" class="sr-only">Password</label>
            <input
              id="password"
              name="password"
              type="password"
              placeholder="Password"
              autocomplete="current-password"
              required="true"
              class="block w-full shadow-sm focus:ring-teal-500 focus:border-teal-500 sm:text-sm border-gray-300 rounded-md"
            />
          </div>

          <div>
            <button
              type="submit"
              class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-teal-600 hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
            >
              Create your account
            </button>
          </div>
        </form>
      </div>
    </div>
    <div class="px-4 py-6 bg-white border-t-2 border-gray-200 sm:px-10">
      <p class="text-xs leading-5 text-gray-500">
        By signing up, you agree to our
        <a href="#" class="font-medium text-gray-900 hover:underline">Terms</a>,
        <a href="#" class="font-medium text-gray-900 hover:underline">Data Policy</a> and
        <a href="#" class="font-medium text-gray-900 hover:underline">Cookies Policy</a>.
      </p>
    </div>
  </div>
</template>

<style scoped></style>
