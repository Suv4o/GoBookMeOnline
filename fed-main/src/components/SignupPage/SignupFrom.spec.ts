import { render, cleanup, fireEvent } from '@testing-library/vue'
import { describe, it, expect, beforeAll, vi } from 'vitest'
import { createRouter, createWebHistory, Router } from 'vue-router'
import { shallowMount } from '@vue/test-utils'
import SignupForm from './SignupForm.vue'
import { initializeFirebaseApp } from '../../config/firebase.config'
import { Auth, getAuth } from 'firebase/auth'
import { pinia } from '../../main'
import { TestUtils } from '../../utils/helpers'
import { useValidator } from '../../utils/composables/validator'
import { default as useStatePhoneVerification } from '../PhoneVerificationPage/useState'
import { useAuthStore } from '../../store/auth'
import { ComponentPublicInstance } from 'vue'

let router: Router
let auth: Auth

describe('SignupForm', async () => {
  beforeAll(() => {
    router = createRouter({
      scrollBehavior(to, from, savedPosition) {
        if (savedPosition) {
          return savedPosition
        } else {
          return { top: 0 }
        }
      },
      history: createWebHistory(),
      routes: [
        {
          path: '/',
          name: 'home',
          components: {},
        },
        {
          path: '/email-verification',
          name: 'email-verification',
          components: {},
        },
        {
          path: '/phone-verification',
          name: 'phone-verification',
          components: {},
        },
      ],
    })
    auth = getAuth(initializeFirebaseApp(process.env as ImportMetaEnv))
  })

  it('render component correctly and create a user with phone, email or google', async () => {
    const wrapper = render(SignupForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    const buttonCreateAUser = wrapper.getByTestId('Create a user')
    const buttonSignupWithGoogle = wrapper.getByTestId('Sign up with Google')

    // Mock signUpWithGoogle() function
    TestUtils.Vitest.mockImplementationClick(buttonSignupWithGoogle, () => undefined)

    // Mock createUser() function
    TestUtils.Vitest.mockImplementationSubmit(buttonCreateAUser, () => undefined)

    await fireEvent.submit(buttonCreateAUser)
    await fireEvent.click(buttonSignupWithGoogle)

    expect(buttonCreateAUser).toBeTruthy()
    expect(buttonCreateAUser).respondTo('submit')

    expect(buttonSignupWithGoogle).toBeTruthy()
    expect(buttonSignupWithGoogle).respondTo('click')

    cleanup()
  })

  it('call createUser() function', async () => {
    const wrapper = shallowMount(SignupForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    const wrapperVm = wrapper.vm as ComponentPublicInstance & {
      createUser: () => void
      signInWithToken: () => void
      sendVerificationEmailLink: () => void
      clearInputs: () => void
      fullName: string
      phoneOrEmail: string
    }

    // Mock and test createUser() and test function
    const spyOnCreateUser = vi.spyOn(wrapperVm, 'createUser')
    spyOnCreateUser.mockImplementation(() => undefined)
    await wrapperVm.createUser()

    expect(spyOnCreateUser).toHaveBeenCalled()

    // Test useValidator() function
    const fullName = 'Foo Bar'
    const incorrectFullName = 'Foo Bar 123'
    const email = 'test@test.com'
    const incorrectEmail = 'test'
    const phone = '+61411111111'

    const testWithFullNameAndEmail = useValidator({
      fullName: fullName,
      phoneOrEmail: email,
    })

    expect(testWithFullNameAndEmail.validProps.fullName.valid).toBe(true)
    expect(testWithFullNameAndEmail.validProps.fullName.message).toBe('')
    expect(testWithFullNameAndEmail.validProps.mobilePhone.valid).toBe(false)
    expect(testWithFullNameAndEmail.validProps.mobilePhone.message).toBe(
      'Please enter a valid Mobile Phone. The phone number must start with a + and must be between 7 and 16 digits.'
    )
    expect(testWithFullNameAndEmail.validProps.email.valid).toBe(true)
    expect(testWithFullNameAndEmail.validProps.email.message).toBe('')
    expect(testWithFullNameAndEmail.validProps.phoneOrEmail.valid).toBe(true)
    expect(testWithFullNameAndEmail.validProps.phoneOrEmail.message).toBe('')

    const testWithFullNameAndPhone = useValidator({
      fullName: fullName,
      phoneOrEmail: phone,
    })

    expect(testWithFullNameAndPhone.validProps.fullName.valid).toBe(true)
    expect(testWithFullNameAndPhone.validProps.fullName.message).toBe('')
    expect(testWithFullNameAndPhone.validProps.mobilePhone.valid).toBe(true)
    expect(testWithFullNameAndPhone.validProps.mobilePhone.message).toBe('')
    expect(testWithFullNameAndPhone.validProps.phoneOrEmail.valid).toBe(true)
    expect(testWithFullNameAndPhone.validProps.phoneOrEmail.message).toBe('')

    const testWithIncorrectFullNameAndIncorrectEmail = useValidator({
      fullName: incorrectFullName,
      phoneOrEmail: incorrectEmail,
    })

    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.fullName.valid).toBe(false)
    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.fullName.message).toBe(
      'Please enter a valid full name.'
    )
    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.mobilePhone.valid).toBe(false)
    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.mobilePhone.message).toBe(
      'Please enter a valid Mobile Phone. The phone number must start with a + and must be between 7 and 16 digits.'
    )
    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.email.valid).toBe(false)
    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.email.message).toBe('Please enter a valid Email.')
    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.phoneOrEmail.valid).toBe(false)
    expect(testWithIncorrectFullNameAndIncorrectEmail.validProps.phoneOrEmail.message).toBe(
      'Please enter a valid Mobile Number or Email. The phone number must start with a + and must be between 7 and 16 digits.'
    )

    // Mock and test signInWithToken() and test function
    const spyOnSignInWithToken = vi.spyOn(wrapperVm, 'signInWithToken')
    spyOnSignInWithToken.mockImplementation(() => undefined)
    await wrapperVm.signInWithToken()

    expect(spyOnSignInWithToken).toHaveBeenCalled()

    // Mock and test sendVerificationEmailLink() and test function
    const spyOnSendVerificationEmailLink = vi.spyOn(wrapperVm, 'sendVerificationEmailLink')
    spyOnSendVerificationEmailLink.mockImplementation(() => undefined)
    await wrapperVm.sendVerificationEmailLink()

    expect(spyOnSendVerificationEmailLink).toHaveBeenCalled()

    // Test clearInputs() function
    const spyOnClearInputs = vi.spyOn(wrapperVm, 'clearInputs')

    await wrapper.find('[data-testid="Full Name"]').setValue('Foo Bar')
    await wrapper.find('[data-testid="Phone or Email"]').setValue('+61411111111')

    expect(wrapperVm.fullName).toBe('Foo Bar')
    expect(wrapperVm.phoneOrEmail).toBe('+61411111111')

    await wrapperVm.clearInputs()

    expect(wrapperVm.fullName).toBe('')
    expect(wrapperVm.phoneOrEmail).toBe('')
    expect(spyOnClearInputs).toHaveBeenCalled()

    // Test setting phone verification useStore()
    await wrapper.find('[data-testid="Full Name"]').setValue('Foo Bar')
    await wrapper.find('[data-testid="Phone or Email"]').setValue('+61411111111')

    const { fullName: name, phoneNumber, setFullName, setPhoneNumber } = useStatePhoneVerification()
    setFullName(wrapperVm.fullName)
    setPhoneNumber(wrapperVm.phoneOrEmail)

    expect(name.value).toBe('Foo Bar')
    expect(phoneNumber.value).toBe('+61411111111')

    spyOnCreateUser.mockReset()
    spyOnSignInWithToken.mockReset()
    spyOnSendVerificationEmailLink.mockReset()
    spyOnClearInputs.mockReset()
  })

  it('call signUpWithGoogle() function', async () => {
    const wrapper = shallowMount(SignupForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    const wrapperVm = wrapper.vm as ComponentPublicInstance & {
      signUpWithGoogle: () => void
      storeUserToDatabase: () => void
    }

    // Mock and test signUpWithGoogle() and test function
    const spyOnSignUpWithGoogle = vi.spyOn(wrapperVm, 'signUpWithGoogle')
    spyOnSignUpWithGoogle.mockImplementation(() => undefined)
    await wrapperVm.signUpWithGoogle()

    expect(spyOnSignUpWithGoogle).toHaveBeenCalled()

    // Mock and test storeUserToDatabase() and test function
    const spyOnStoreUserToDatabase = vi.spyOn(wrapperVm, 'storeUserToDatabase')
    // Set the auth store with pinia
    const useStoreAuth = useAuthStore(pinia)
    spyOnStoreUserToDatabase.mockImplementation(() => {
      useStoreAuth.user = {
        uid: 'ZU4jn14tbWfspKuWorshlqQ1s2E3',
        firstName: 'Foo',
        lastName: 'Bar',
        role: 'USER_DEFAULT',
        email: 'test@test.com',
        phoneNumber: null,
        displayName: 'Foo Bar',
        photoURL: null,
        emailVerified: true,
      }
    })
    await wrapperVm.storeUserToDatabase()

    expect(spyOnStoreUserToDatabase).toHaveBeenCalled()
    expect(useStoreAuth.user).toEqual({
      uid: 'ZU4jn14tbWfspKuWorshlqQ1s2E3',
      firstName: 'Foo',
      lastName: 'Bar',
      role: 'USER_DEFAULT',
      email: 'test@test.com',
      phoneNumber: null,
      displayName: 'Foo Bar',
      photoURL: null,
      emailVerified: true,
    })
  })

  it('snap shot matches', async () => {
    const wrapper = render(SignupForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
