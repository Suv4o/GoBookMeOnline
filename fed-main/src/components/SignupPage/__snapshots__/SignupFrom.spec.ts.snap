// Vitest Snapshot v1

exports[`SignupForm > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="bg-white sm:max-w-md sm:w-full sm:mx-auto sm:rounded-lg sm:overflow-hidden"
      >
        <div
          class="px-4 py-8 sm:px-10"
        >
          <div>
            <p
              class="text-sm font-medium text-gray-700"
            >
              Sign up with Google
            </p>
            <div
              class="mt-1 grid grid-cols-1 gap-3"
            >
              <div>
                <a
                  class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
                  data-testid="Sign up with Google"
                  href="javascript:;"
                >
                  <span
                    class="sr-only"
                  >
                    Sign in with Google
                  </span>
                  <svg
                    aria-hidden="true"
                    class="w-5 h-5"
                    fill="currentColor"
                    viewBox="0 0 210 210"
                  >
                    <path
                      d="M0,105C0,47.103,47.103,0,105,0c23.383,0,45.515,7.523,64.004,21.756l-24.4,31.696C133.172,44.652,119.477,40,105,40
	c-35.841,0-65,29.159-65,65s29.159,65,65,65c28.867,0,53.398-18.913,61.852-45H105V85h105v20c0,57.897-47.103,105-105,105
	S0,162.897,0,105z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
          <div
            class="mt-6 relative"
          >
            <div
              aria-hidden="true"
              class="absolute inset-0 flex items-center"
            >
              <div
                class="w-full border-t border-gray-300"
              />
            </div>
            <div
              class="relative flex justify-center text-sm"
            >
              <span
                class="px-2 bg-white text-gray-500"
              >
                 Or 
              </span>
            </div>
          </div>
          <div
            class="mt-6"
          >
            <form
              class="space-y-6"
              data-testid="Create a user"
            >
              <div>
                <label
                  class="sr-only"
                  for="name"
                >
                  Full Name
                </label>
                <input
                  autocomplete="name"
                  class="block w-full shadow-sm sm:text-sm border-gray-300 rounded-md focus:ring-teal-500 focus:border-teal-500"
                  data-testid="Full Name"
                  id="name"
                  name="name"
                  placeholder="Full Name"
                  required=""
                  type="text"
                />
                <!--v-if-->
              </div>
              <div>
                <label
                  class="sr-only"
                  for="mobile-or-email"
                >
                  Mobile Number or Email
                </label>
                <input
                  autocomplete="email|tel"
                  class="block w-full shadow-sm sm:text-sm border-gray-300 rounded-md focus:ring-teal-500 focus:border-teal-500"
                  data-testid="Phone or Email"
                  id="mobile-or-email"
                  name="mobile-or-email"
                  placeholder="Mobile Number or Email"
                  required=""
                  type="text"
                />
                <!--v-if-->
              </div>
              <div>
                <button
                  class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-teal-600 hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                  type="submit"
                >
                  <!--v-if-->
                   Create your account
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="px-4 py-6 bg-white border-t-2 border-gray-200 sm:px-10"
        >
          <p
            class="text-xs leading-5 text-gray-500"
          >
             By signing up, you agree to our 
            <a
              class="font-medium text-gray-900 hover:underline"
              href="#"
            >
              Terms
            </a>
            , 
            <a
              class="font-medium text-gray-900 hover:underline"
              href="#"
            >
              Data Policy
            </a>
             and 
            <a
              class="font-medium text-gray-900 hover:underline"
              href="#"
            >
              Cookies Policy
            </a>
            . 
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white sm:max-w-md sm:w-full sm:mx-auto sm:rounded-lg sm:overflow-hidden"
    >
      <div
        class="px-4 py-8 sm:px-10"
      >
        <div>
          <p
            class="text-sm font-medium text-gray-700"
          >
            Sign up with Google
          </p>
          <div
            class="mt-1 grid grid-cols-1 gap-3"
          >
            <div>
              <a
                class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
                data-testid="Sign up with Google"
                href="javascript:;"
              >
                <span
                  class="sr-only"
                >
                  Sign in with Google
                </span>
                <svg
                  aria-hidden="true"
                  class="w-5 h-5"
                  fill="currentColor"
                  viewBox="0 0 210 210"
                >
                  <path
                    d="M0,105C0,47.103,47.103,0,105,0c23.383,0,45.515,7.523,64.004,21.756l-24.4,31.696C133.172,44.652,119.477,40,105,40
	c-35.841,0-65,29.159-65,65s29.159,65,65,65c28.867,0,53.398-18.913,61.852-45H105V85h105v20c0,57.897-47.103,105-105,105
	S0,162.897,0,105z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="mt-6 relative"
        >
          <div
            aria-hidden="true"
            class="absolute inset-0 flex items-center"
          >
            <div
              class="w-full border-t border-gray-300"
            />
          </div>
          <div
            class="relative flex justify-center text-sm"
          >
            <span
              class="px-2 bg-white text-gray-500"
            >
               Or 
            </span>
          </div>
        </div>
        <div
          class="mt-6"
        >
          <form
            class="space-y-6"
            data-testid="Create a user"
          >
            <div>
              <label
                class="sr-only"
                for="name"
              >
                Full Name
              </label>
              <input
                autocomplete="name"
                class="block w-full shadow-sm sm:text-sm border-gray-300 rounded-md focus:ring-teal-500 focus:border-teal-500"
                data-testid="Full Name"
                id="name"
                name="name"
                placeholder="Full Name"
                required=""
                type="text"
              />
              <!--v-if-->
            </div>
            <div>
              <label
                class="sr-only"
                for="mobile-or-email"
              >
                Mobile Number or Email
              </label>
              <input
                autocomplete="email|tel"
                class="block w-full shadow-sm sm:text-sm border-gray-300 rounded-md focus:ring-teal-500 focus:border-teal-500"
                data-testid="Phone or Email"
                id="mobile-or-email"
                name="mobile-or-email"
                placeholder="Mobile Number or Email"
                required=""
                type="text"
              />
              <!--v-if-->
            </div>
            <div>
              <button
                class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-teal-600 hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                type="submit"
              >
                <!--v-if-->
                 Create your account
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="px-4 py-6 bg-white border-t-2 border-gray-200 sm:px-10"
      >
        <p
          class="text-xs leading-5 text-gray-500"
        >
           By signing up, you agree to our 
          <a
            class="font-medium text-gray-900 hover:underline"
            href="#"
          >
            Terms
          </a>
          , 
          <a
            class="font-medium text-gray-900 hover:underline"
            href="#"
          >
            Data Policy
          </a>
           and 
          <a
            class="font-medium text-gray-900 hover:underline"
            href="#"
          >
            Cookies Policy
          </a>
          . 
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
